geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-orc/src/main/scala/org/locationtech/geomesa/fs/storage/orc/OrcFileSystemStorage.scala

```diff
@@ -81,11 +83,23 @@ object OrcFileSystemStorage {
     * @return
     */
   def fieldCount(sft: SimpleFeatureType, fid: Boolean = true): Int = {
-    import scala.collection.JavaConversions._
-    sft.getAttributeDescriptors.count(d => classOf[Geometry].isAssignableFrom(d.getType.getBinding)) +
-        sft.getAttributeCount + (if (fid) { 1 } else { 0 })
+    val attributes = sft.getAttributeDescriptors.asScala.foldLeft(0) { case (sum, d) => sum + fieldCount(d) }
+    if (fid) { attributes + 1 } else { attributes }
   }
 
+  /**
+   * Gets a count of the Orc fields created for a given attribute
+   *
+   * @param descriptor descriptor
+   * @return
+   */
+  def fieldCount(descriptor: AttributeDescriptor): Int = {
+    descriptor.getType.getBinding match {
+      // plain Geometry bindings are encoded in a single WKB column, others as x + y
+      case b if classOf[Geometry].isAssignableFrom(b) && b != classOf[Geometry] => 2
```

Comment:
If someone had a Simple Feature Collection with a GeometryCollection (instead of 'Geometry' as the type), would things work?

Reply:
no... i noticed that we don't handle geometry collections in several different places. i figured that there wasn't really any benefit to declaring the type of GeometryCollection vs just using Geometry, so I didn't bother to go through and update them. we could though.